channeldb/invoice_test.go

```diff
@@ -578,6 +578,67 @@ func TestDuplicateSettleInvoice(t *testing.T) {
 	}
 }
 
+// TestFetchAllInvoices tests that FetchAllInvoices works as expected.
+func TestFetchAllInvoices(t *testing.T) {
+	t.Parallel()
+
+	db, cleanUp, err := makeTestDB()
+	defer cleanUp()
+	if err != nil {
+		t.Fatalf("unable to make test db: %v", err)
+	}
+
+	const numInvoices = 100
+	contractStates := []ContractState{
+		ContractOpen, ContractSettled, ContractCanceled, ContractAccepted,
+	}
+
+	var expectedPendingInvoices []Invoice
+	var expectedAllInvoices []Invoice
+
+	for i := 1; i < numInvoices+1; i++ {
+		invoiceValue := lnwire.MilliSatoshi(i)
+		invoice, err := randInvoice(invoiceValue)
+		invoice.AddIndex = uint64(i)
+
+		if err != nil {
+			t.Fatalf("unable to create invoice: %v", err)
+		}
+
+		invoice.State = contractStates[mrand.Intn(len(contractStates))]
```

Comment:
Random in unit tests is risky. For a time it is unlikely to impact the execution flow, but for this the non-deterministically-ness gets more serious.

Reply:
Removed the random in favor of mod.